src/main/java/net/jodah/failsafe/SyncFailsafe.java

```diff
@@ -114,6 +115,27 @@ public void run(ContextualRunnable runnable) {
   }
 
   /**
+   * Creates a proxy instance of the provided {@link T} instance such that all public methods
```

Comment:
It's not all public methods, it's all methods of T.

Reply:
Well, since we are only proxying interfaces, its only the public methods.

Reply:
Given that there's some discussion about allowing private methods on interfaces in Java 9, I think it would be safer to say _all methods of interface T_. I don't know that dynamic proxies would in fact handle private method invocations, but since you aren't checking to see whether the invocation is of a public method currently, better safe than sorry.

Reply:
By virtue of the type <T> being an interface Class, it *must* be public methods.
In fact, the return type is the interface type, which means you can *only* call public methods.

I found online 
`As of Java 8, interfaces can have default methods, and as of Java 9, an interface will be allowed to have a private methods which can only be accessed by default methods in the same interface.`
(that means the only visible interface from a call of `proxy` is the public methods)

Reply:
Lets not bike-shed -- i'll change it to "all methods" if its blocking the review :)

Reply:
I guess it struck me as odd to qualify with _public_ when the _public-ness_ is just a by-product of the fact that dynamic proxies currently only wrap interface types. Saying _all public methods_ is correct, so I won't make any further objection, but I reacted because it felt like an unnecessary additional detail.

Your call.